lifting-state-up/passing-down-event-handlers.md

```diff
@@ -0,0 +1,250 @@
+| `studentData.map(...)`                              | We'll map over `studentData`...                                                                                                                                      |
+| `student => {...}`                                 | Inside `map`, we'll reference each item with `student`...                                                                                                            |
+| `if (student.id === updatedStudent.id) {`           | If we found the student that our `updatedStudent` is updating...                                                                                                     |
+| `return updatedStudent`                             | The item we should map to our `students` array is the updated student data!                                                                                          |
+| `else { return student; }`                          | Otherwise, we should map the unchanged `student` into `students`.                                                                                                    |
+| `setStudentData(students);`                         | Ultimately, we want to update the `studentData` on state. We use our state update function, `setStudentData`, and we update it to our newly formed `students` array. |
+
+Our `App` now defines and holds a function that can update our `studentData`, named `updateStudentData`.
+
+## Send This Function to `StudentList`
+
+Now that we've defined `updateStudentData`, imagine if other components could _use_ this function.
+
+If other components could use this function, they'd have the ability to update the `studentData` in `App`! We want _the `Student` component_ to update the `studentData` in `App`.
+
+Other components _can_ use the `updateStudentData` function if they can reference it... and they can, _using props_.
+
+First, let's change `App` so that it sends this function to `StudentList` through a new prop named `onUpdateStudent`.
+
+<!-- prettier-ignore-start -->
+```js
+  return (
+    <main>
+      <h1>Attendance</h1>
+      <StudentList
+        students={studentData}
+        onUpdateStudent={updateStudentData}
+      ></StudentList>
+    </main>
+  );
+```
+<!-- prettier-ignore-end -->
+
+Let's update the PropTypes for `StudentList`. Our `StudentList` component now expects a `prop` named `onUpdateStudent`, whose value is a _function_.
+
+<!-- prettier-ignore-start -->
+```js
+StudentList.propTypes = {
+    students: PropTypes.arrayOf(PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        nameData: PropTypes.string.isRequired,
+        emailData: PropTypes.string.isRequired,
+        isPresentData: PropTypes.bool
+    })),
+    onUpdateStudent: PropTypes.func
```

Comment:
Since this winds up being set as required in Student later on, we should probably mark it required here as well
```suggestion
    onUpdateStudent: PropTypes.func.isRequired
```